crates/doc/src/inference.rs

```diff
@@ -1484,6 +1578,182 @@ impl Shape {
     }
 }
 
+const MAX_ROOT_FIELDS: usize = 750;
+const MAX_NESTED_FIELDS: usize = 200;
+
+impl Shape {
+    // Widen a Shape to make the provided AsNode value fit.
+    // Returns a hint if some locations might exceed their maximum allowable size.
+    // NOTE: If a particular location defines `additionalProperties` as a subschema, don't
```

Comment:
I actually decided to move the docstring from `ObjShape::widen` here instead since this is a public function so it'll be more visible and explains what's going on better.